littleblog/entry/serializers.py

```diff
@@ -1,15 +1,36 @@
 from rest_framework import serializers
 from .models import Entry, UserProfile
 from django.contrib.auth.models import User
+from rest_framework.validators import UniqueValidator
+from django.contrib.auth.password_validation import UserAttributeSimilarityValidator, MinimumLengthValidator, \
+    CommonPasswordValidator, NumericPasswordValidator
 
 
 class UserProfileCreateSerializer(serializers.ModelSerializer):
 
-    user = serializers.PrimaryKeyRelatedField(queryset=User.objects.all())
+    UserProfile = serializers.ReadOnlyField(source='profile.nickname')
+    nickname = serializers.CharField(max_length=50, required=True, write_only=True)
+    avatar = serializers.ImageField(required=False, write_only=True)
```

Comment:
Why write_only? There's nothing wrong to return these fields in response.

```suggestion
    nickname = serializers.CharField(source="profile.name")
    avatar = serializers.ImageField(source="profile.avatar", required=False)
```

you don't need `UserProfile = ` thing

Reply:
Crap...I was just mistaken in my suggestion that we can't pass the values to fields which do not belong to the serializer class that was mentioned in Meta.model. So if we use them source parameter we can both fetch and pass, didn't know that.